File: src/main/kotlin/info/hildegynoid/cue/converter/entity/CueTrackInfo.kt
package info.hildegynoid.cue.converter.entity

import info.hildegynoid.cue.converter.loader.EOL
import info.hildegynoid.cue.converter.loader.INDENT
import info.hildegynoid.cue.converter.loader.escape

data class CueTrackInfo(
    var number: Int,
    var title: String = "",
    var performer: String = "",
    var fileName: String = "",
    var indices: MutableMap<Int, CueTrackIndex> = mutableMapOf()
) {
    override fun toString(): String {
        return String.format("TRACK %02d AUDIO", number) + EOL +
            INDENT + "TITLE \"${title.escape()}\"" + EOL +
            INDENT + "PERFORMER \"${performer.escape()}\"" + EOL +
            indices.map { INDENT + String.format("INDEX %02d", it.key) + " " + it.value + EOL }
                .joinToString("")
    }
}

File: src/cue/entity/CueTrackInfo.ts
import { CueTrackIndex } from './CueTrackIndex';
import { EOL, escapeString, INDENT, zeroFill } from '../Utils';

export class CueTrackInfo {
  number: number;

  title: string;

  performer: string;

  fileName: string;

  indices: Map<number, CueTrackIndex> = new Map();

  constructor(
    number: number,
    title?: string,
    performer?: string,
    fileName?: string
  ) {
    this.number = number;
    this.title = title ?? '';
    this.performer = performer ?? '';
    this.fileName = fileName ?? '';
  }

  getTrack(): string {
    return `${this.number}.${this.performer} - ${this.title}`;
  }

  toString(): string {
    let indices: string = '';
    this.indices.forEach((value, key) => {
      indices += `${INDENT}INDEX ${zeroFill(key)} ${value}${EOL}`;
    });

    return `TRACK ${zeroFill(
      this.number
    )} AUDIO${EOL}${INDENT}TITLE "${escapeString(
      this.title
    )}"${EOL}${INDENT}PERFORMER "${escapeString(
      this.performer
    )}"${EOL}${indices}`;
  }
}
